client/src/main/java/org/apache/nemo/client/JobLauncher.java

```diff
@@ -261,7 +264,7 @@ public static void launchDAG(final IRDAG dag,
                                final Map<Serializable, Object> broadcastVariables,
                                final String jobId) {
     // launch driver if it hasn't been already
-    if (driverReadyLatch == null) {
+    if (driverReadyLatch == null || jobDoneLatch == null) {
```

Comment:
This code is hard to understand. Isn't there a variable for the driver status? and can't we use the variable for checking the driver status?

Reply:
This statement simply checks whether the `setup` method has been called or not, and calls it if it haven't. The driverReadyLatch has to be there to wait for the driver to be ready later on in the code. Do you think that it would be better to create another `boolean` variable to indicate whether the `setup` method has been called?

Reply:
Yeah I think using the bool value is more readable than checking null.

Reply:
I've reflected the corresponding change on the branch. Please check!